pkg/doc/verifiable/credential_jwt.go

```diff
@@ -0,0 +1,260 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package verifiable
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/square/go-jose/v3"
+	"github.com/square/go-jose/v3/jwt"
+)
+
+// JWTCredClaims is JWT Claims extension by Verifiable Credential.
+type JWTCredClaims struct {
+	*jwt.Claims
+
+	Credential *rawCredential `json:"vc,omitempty"`
+}
+
+// SerializeSigned serialized JWT into signed form (JWT)
+func (vcc *JWTCredClaims) SerializeSigned(signatureAlg JWTAlgorithm, privateKey interface{}, keyID string) (string, error) { //nolint:lll
+	key := jose.SigningKey{Algorithm: signatureAlg.Jose(), Key: privateKey}
+
+	var signerOpts = &jose.SignerOptions{}
+	signerOpts.WithType("JWT")
+	signerOpts.WithHeader("kid", keyID)
+
+	signer, err := jose.NewSigner(key, signerOpts)
+	if err != nil {
+		return "", fmt.Errorf("failed to create signer: %w", err)
+	}
+
+	// create an instance of Builder that uses the rsa signer
+	builder := jwt.Signed(signer).Claims(vcc)
+
+	// validate all ok, sign with the RSA key, and return a compact JWT
+	rawJWT, err := builder.CompactSerialize()
+	if err != nil {
+		return "", fmt.Errorf("failed to sign JWT: %w", err)
+	}
+
+	return rawJWT, nil
+}
+
+// SerializeUnsecured serialized JWT into unsecured JWT.
+func (vcc *JWTCredClaims) SerializeUnsecured() (string, error) {
```

Comment:
I think the term Marshal has been used more frequently in this project.

Reply:
Renamed.